suite-native/module-add-accounts/src/hooks/useAddCoinAccount.ts

```diff
@@ -92,100 +119,180 @@ export const useAddCoinAccount = () => {
+                showTooManyAccountsAlert();
+                return false;
+            }
 
-        // Do not allow adding another empty account if there is already one
-        const emptyAccounts = currentAccountTypeAccounts.filter(account => account.empty);
+            // Do not allow adding another empty account if there is already one
+            const emptyAccounts = currentAccountTypeAccounts.filter(account => account.empty);
 
-        if (emptyAccounts.length > 0) {
-            showAnotherEmptyAccountAlert();
-            return false;
-        }
+            if (emptyAccounts.length > 0) {
+                showAnotherEmptyAccountAlert();
+                return false;
+            }
+            return true;
+        },
+        [accounts, getDefaultAccountType, showAnotherEmptyAccountAlert, showTooManyAccountsAlert],
+    );
+
+    const addCoinAccount = useCallback(
+        async ({ network, accountType }: NetworkWithAccountType) => {
+            if (!device?.state) {
+                setNetworkWithTypeToAdd(undefined);
+                return false;
+            }
+
+            const selectedType = accountType ?? getDefaultAccountType({ network });
 
-        setIsAddingCoinAccount(true);
-        const account = await dispatch(
-            addAndDiscoverNetworkAccountThunk({
+            const canAddAccount = checkCanAddAccount({
                 network,
-                accountType,
-                deviceState: device.state,
-            }),
-        ).unwrap();
+                accountType: selectedType,
+            });
+
+            if (!canAddAccount) {
+                return false;
+            }
 
-        setIsAddingCoinAccount(false);
+            setIsAddingCoinAccount(true);
+            const account = await dispatch(
+                addAndDiscoverNetworkAccountThunk({
+                    network,
+                    accountType: selectedType,
+                    deviceState: device.state,
+                }),
+            ).unwrap();
 
-        if (account) {
-            navigation.navigate(RootStackRoutes.AccountDetail, {
-                accountKey: account.key,
-                tokenContract: undefined,
+            setIsAddingCoinAccount(false);
+            setNetworkWithTypeToAdd(undefined);
+            if (account) {
+                navigation.dispatch(
+                    CommonActions.reset({
+                        index: 0,
+                        routes: [
+                            {
+                                name: RootStackRoutes.AppTabs,
+                                params: {
+                                    screen: AppTabsRoutes.AccountsStack,
+                                },
+                            },
+                        ],
+                    }),
+                );
+                navigation.navigate(RootStackRoutes.AccountDetail, {
+                    accountKey: account.key,
+                    tokenContract: undefined,
+                });
```

Comment:
Why are you reseting to then navigate?

Reply:
to go back to root after going from the account detail and not via the other screens

Reply:
I think it's redundant. You can just call `navigation.navigate`

Reply:
I'll keep it this way for now and revisit it later in following PRs, where I need to make it work from multiple stacks (account | receive)

Reply:
5c5f2af2c4